pkg/apis/harvesterhci.io/v1beta1/upgrade.go

```diff
@@ -67,3 +75,10 @@ type NodeUpgradeStatus struct {
 	Reason  string `json:"reason,omitempty"`
 	Message string `json:"message,omitempty"`
 }
+
+// GetUpgradeImage returns the upgrade image as a Docker image reference, e.g. `rancher/harvester-upgrade:latest`.
+// If the `spec.upgradeImage` field is not provided, it constructs the image reference using the provided repository and tag.
+func (u *Upgrade) GetUpgradeImage(repository, tag string) string {
+	defaultValue := fmt.Sprintf("%s:%s", repository, tag)
```

Comment:
given this feature is more for the convenience of develop,

when the target tag is a formal released version, e.g. v1.6.0, then it is better to skip the input

Reply:
There are alway users trying to use every possible options and report bugs; adding this to configuration might bring some potential troubles to us.

Even for dev engineers, if the installer-ISO and the customized image has big differences, the ugprade might also fail and needs more time to dig.